scripts/dump-openapi.py

```diff
@@ -44,17 +67,95 @@ def prefix_absolute_path_references(text, base_url):
     """
     return text.replace("](/", "]({}/".format(base_url))
 
-def edit_links(node, base_url):
-    """Finds description nodes and makes any links in them absolute."""
+def replace_match(text, match, replacement):
+    """Replaces the regex match by the replacement in the text."""
+    return text[:match.start()] + replacement + text[match.end():]
```

Comment:
it might be worth noting that `text` is also accessible as `match.string` here, so you could avoid threading `text` down the call stack. I'm not entirely sure if that makes things clearer though!

Reply:
Indeed, I changed the arguments to only take the `match` and `replacement`.